tools/shared.py

```diff
@@ -577,7 +577,8 @@ def check_sanity(force=False):
       return # config stored directly in EM_CONFIG => skip sanity checks
     expected = generate_sanity()
 
-    sanity_file = Cache.get_path('sanity.txt')
+    temp_cache = cache.Cache(CACHE, use_subdir=False)
```

Comment:
this worries me a little. are there no side effects to creating another `Cache` object?

how about just doing `os.path.dirname(..)` to go up one level?

Reply:
Meanwhile I see that's how the vanilla file already works, so this doesn't add any risk.

Reply:
Yeah.. i just do the same as the vanilla file.    We might want to clean this up at some point.